bftengine/src/bftengine/messages/ClientRequestMsg.cpp

```diff
@@ -150,6 +150,12 @@ void ClientRequestMsg::validateImp(const ReplicasInfo& repInfo) const {
       }
     }
   }
+  if ((isClientTransactionSigningEnabled == false) && (header->flags & RECONFIG_FLAG)) {
```

Comment:
2 comments:
1) The validation of the client ID is very important here. I might not fully understandf the code, but do you ignore isIdOfExternalClient on purpose?
2) I think this statement  should start with an else.
I would refactor this part with the above:
`if (isIdOfExternalClient && isClientTransactionSigningEnabled) {`
Change to
`if (iisClientTransactionSigningEnabled) {`
and have here an else with the rest of the applied logic refactored accordingly.
Else you are always leave the chance to override previous set to true as part of future changes.

Reply:
fixed